src/supervisor/watchers/handlers/deployment-config.ts

```diff
@@ -0,0 +1,32 @@
+import { deleteWorkload } from './workload';
+import { WorkloadKind } from '../../types';
+import { FALSY_WORKLOAD_NAME_MARKER, V1DeploymentConfig } from './types';
+
+export async function deploymentConfigWatchHandler(
+  deploymentConfig: V1DeploymentConfig,
+): Promise<void> {
+  if (
+    !deploymentConfig.metadata ||
+    !deploymentConfig.spec ||
+    !deploymentConfig.spec.template.metadata ||
+    !deploymentConfig.spec.template.spec ||
+    !deploymentConfig.status
+  ) {
+    return;
+  }
+
+  const workloadName =
+    deploymentConfig.metadata.name || FALSY_WORKLOAD_NAME_MARKER;
+
+  await deleteWorkload(
```

Comment:
how come the workload is deleted?

Reply:
oh it only has a delete command!

Reply:
should `delete` be in the name of the handler? (I see that this is not what we have done in the others apart from pod but maybe it should be for consistency?)

Reply:
@kat1906 Yes I think it's the second time the names trip us up here, I'll open a follow up PR to improve them 🙏 The `deleteWorkload()` here only sends an event to Upstream that this workload was deleted, it doesn't actually do anything to the workload in K8s 😄